jxl/src/entropy_coding/owned_histograms.rs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) the JPEG XL Project Authors. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+use std::rc::Rc;
+
+use crate::{bit_reader::BitReader, error::Result};
+
+use super::decode::Histograms;
+
+/// Equivalent to a `crate::entropy_coding::Reader`, but owns the histograms.
+pub struct Reader {
+    // Safety note: 'static is not correct. `inner` actually borrows from `*histograms`.
+    // Since `inner` is never exposed outside this struct, we can guarantee that `inner` is always
+    // dropped before `histograms`.
+    //
+    // Drop order note: the order of these fields guarantees that `inner` is dropped before
+    // `histograms`.
+    inner: super::decode::Reader<'static>,
```

Comment:
Include some discussion on why unsafety is justified here, e.g. brief discussion of alternatives and drawbacks thereof.